src/app/ReviewService.Abstractions/ReviewCondition.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace ReviewService.Abstractions;
+
+/// <summary>
+/// Provides common review prompt conditions.
+/// </summary>
+public static class ReviewCondition
+{
+	/// <summary>
+	/// The number of completed primary actions must be at least <paramref name="minimumActionCompleted"/>.
+	/// </summary>
+	/// <param name="minimumActionCompleted">The minimum number of completed actions.</param>
+	/// <returns>Review prompt condition which can be used to determine whether the user as completed enough primary actions.</returns>
+	public static ReviewConditionDelegate PrimaryActionCompletedAtLeast(int minimumActionCompleted) =>
+		(ct, reviewPromptSettings, currentDateTime) => Task.FromResult(reviewPromptSettings.PrimaryActionCompletedCount >= minimumActionCompleted);
+
+	/// <summary>
+	/// The number of completed secondary actions must be at least <paramref name="minimumActionCompleted"/>.
+	/// </summary>
+	/// <param name="minimumActionCompleted">The minimum number of completed actions.</param>
+	/// <returns>Review prompt condition which can be used to determine whether the user as completed enough primary actions.</returns>
+	public static ReviewConditionDelegate SecondaryActionCompletedAtLeast(int minimumActionCompleted) =>
+		(ct, reviewPromptSettings, currentDateTime) => Task.FromResult(reviewPromptSettings.SecondaryActionCompletedCount >= minimumActionCompleted);
+
+	/// <summary>
+	/// The number of times the application has been launched must be at least <paramref name="minimumApplicationLaunched"/>.
+	/// </summary>
+	/// <param name="minimumApplicationLaunched">The minimum number of times the application has been launched.</param>
+	/// <returns>Review prompt condition which can be used to determine whether the user as launched the application enough times.</returns>
+	public static ReviewConditionDelegate ApplicationLaunchedAtLeast(int minimumApplicationLaunched) =>
+		(ct, reviewPromptSettings, currentDateTime) => Task.FromResult(reviewPromptSettings.ApplicationLaunchCount >= minimumApplicationLaunched);
```

Comment:
Maybe I don't see the whole picture, but it seems that we're forcing async here even though we're doing simple if check

Reply:
Yes we are but I don't know if a condition could need it eventually that's why I "forced" it. Since it's like a method that could do anything. Someone might need to do an async thing eventually. But I'm not sure.

Reply:
Well it's using `Task.FromResult` so it's not async.

Reply:
I figured that we could still manually cancel it by using `ct.Register` https://learn.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken.register?view=net-7.0. But it may be overkill